src/main/java/org/broadinstitute/hellbender/utils/test/VariantContextTestUtils.java

```diff
@@ -22,14 +24,40 @@
 import org.testng.Assert;
 
 import java.util.*;
-import java.util.regex.Pattern;
 import java.util.stream.Collectors;
-import java.util.stream.Stream;
 
 public final class VariantContextTestUtils {
 
     private VariantContextTestUtils() {}
 
+    /**
+     * Reads an entire VCF into memory, returning both its VCFHeader and all VariantContext records in
+     * the vcf. Supports both local files and NIO-supported remote filesystems such as GCS.
+     *
+     * For unit/integration testing purposes only! Do not call this method from actual tools!
+     *
+     * @param vcfPath path or URI to a VCF, as a String
+     * @return A Pair with the VCFHeader as the first element, and a List of all VariantContexts from the VCF
+     *         as the second element
+     */
+    public static Pair<VCFHeader, List<VariantContext>> readEntireVCFIntoMemory(final String vcfPath) {
+        Utils.nonNull(vcfPath);
+
+        try ( final FeatureDataSource<VariantContext> vcfReader = new FeatureDataSource<>(vcfPath) ) {
```

Comment:
What happens if it fails during the try instantiation of this reader when it points to a non-existant file? Does it throw an IOException if it can't read the file in the first place because of a bad path?

Reply:
That case is handled by `FeatureDataSource` itself (it throws a `UserException.CouldNotReadInputFile`). I'll add a quick test to cover that case.